packages/integration-sdk-runtime/src/storage/FileSystemGraphObjectStore/indices.ts

```diff
@@ -4,61 +4,49 @@ import {
   GraphObjectIteratee,
   IntegrationError,
 } from '@jupiterone/integration-sdk-core';
-import { FlushedEntityData, FlushedRelationshipData } from '../types';
-import {
-  readJsonFromPath,
-  walkDirectory,
-  WalkDirectoryIterateeInput,
-} from '../../fileSystem';
+import { readJsonFromPath, WalkDirectoryIterateeInput } from '../../fileSystem';
 
 import { buildIndexDirectoryPath } from './path';
+import { iterateParsedGraphFiles } from '../..';
 
 interface IterateIndexInput<GraphObject> {
   type: string;
   iteratee: GraphObjectIteratee<GraphObject>;
 }
 
-export async function iterateEntityTypeIndex<T extends Entity = Entity>({
+interface BaseIterateIndexInput<GraphObject>
```

Comment:
1. What do you think of `IterateCollectionIndexInput`? I suppose that `Base` to me that in a type hierarchy, this would be the type with the minimal number of properties and other types would build on this, but here we see that it is extending something with fewer properties.
2. What do you think about, in cases where function parameters are received as an object for destructuring, we use the name `IterateCollectionIndexParams` instead of `Input`? I prefer `Params` since it reflects the fact that we're creating a type to represent parameters to a function.

Reply:
1. Thanks for the input. I meant to flip the naming the other way. 😬 
2. Definitely! Will change.

Thanks for the great suggestions.